android/src/ti/map/TiClusterRenderer.java

```diff
@@ -0,0 +1,75 @@
+package ti.map;
+
+import com.google.maps.android.clustering.view.DefaultClusterRenderer;
+import com.google.android.gms.maps.GoogleMap;
+import android.content.Context;
+import com.google.maps.android.clustering.ClusterManager;
+import com.google.android.gms.maps.model.Marker;
+import org.appcelerator.titanium.view.TiDrawableReference;
+import android.graphics.Bitmap;
+import com.google.android.gms.maps.model.MarkerOptions;
+import org.appcelerator.titanium.TiApplication;
+import com.google.android.gms.maps.model.BitmapDescriptorFactory;
+import android.view.View;
+import org.appcelerator.titanium.TiDimension;
+import org.appcelerator.titanium.TiPoint;
+import java.util.HashMap;
+import org.appcelerator.titanium.TiC;
+
+public class TiClusterRenderer extends DefaultClusterRenderer<TiClusterMarker> {
+
+    private static final String defaultIconImageHeight = "40dip"; //The height of the default marker icon
+    private static final String defaultIconImageWidth = "36dip"; //The width of the default marker icon
+    private int iconImageHeight = 0;
+    private int iconImageWidth = 0;
+    private ClusterManager<TiClusterMarker> clusterManager;
+
+    public TiClusterRenderer(Context context, GoogleMap map, ClusterManager<TiClusterMarker> clusterManager) {
+        super(context, map, clusterManager);
+        this.clusterManager = clusterManager;
+    }
+
+    @Override
+    protected void onBeforeClusterItemRendered(TiClusterMarker clusterItem, MarkerOptions markerOptions) {
+        AnnotationProxy anno = clusterItem.getProxy();
+
+        if (anno.hasProperty(TiC.PROPERTY_IMAGE)) {
+            TiDrawableReference imageref = TiDrawableReference.fromUrl(anno, (String) anno.getProperty(TiC.PROPERTY_IMAGE));
```

Comment:
Maybe we can use the same image handling as in `AnnotationProxy.handleImage()` to support multiple sources?

Reply:
done, it works now with images and blobs